SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUpstreamChecker.java

```diff
@@ -0,0 +1,81 @@
+package com.hubspot.singularity.scheduler;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.google.common.base.Optional;
+import com.hubspot.baragon.models.BaragonRequest;
+import com.hubspot.baragon.models.BaragonService;
+import com.hubspot.baragon.models.UpstreamInfo;
+import com.hubspot.singularity.LoadBalancerRequestType;
+import com.hubspot.singularity.LoadBalancerRequestType.LoadBalancerRequestId;
+import com.hubspot.singularity.SingularityDeploy;
+import com.hubspot.singularity.SingularityLoadBalancerUpdate;
+import com.hubspot.singularity.SingularityRequest;
+import com.hubspot.singularity.SingularityRequestWithState;
+import com.hubspot.singularity.SingularityTask;
+import com.hubspot.singularity.SingularityTaskId;
+import com.hubspot.singularity.data.DeployManager;
+import com.hubspot.singularity.data.RequestManager;
+import com.hubspot.singularity.data.TaskManager;
+import com.hubspot.singularity.hooks.LoadBalancerClientImpl;
+
+@Singleton
+public class SingularityUpstreamChecker {
+
+  private final LoadBalancerClientImpl lbClient;
+  private final TaskManager taskManager;
+  private final RequestManager requestManager;
+  private final DeployManager deployManager;
+
+  @Inject
+  public SingularityUpstreamChecker(LoadBalancerClientImpl lbClient, TaskManager taskManager, RequestManager requestManager, DeployManager deployManager) {
+    this.lbClient = lbClient;
+    this.taskManager = taskManager;
+    this.requestManager = requestManager;
+    this.deployManager = deployManager;
+  }
+
+  private List<SingularityTask> getActiveSingularityTasksForRequest (String requestId) {
```

Comment:
same formatting nit

Reply:
There is a `public Map<SingularityTaskId, SingularityTask> getTasks(Iterable<SingularityTaskId> taskIds) {` method in `TaskManager` that will make this much easier. It will return the SingularityTask object for each of the provided task ids